exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -89,22 +89,36 @@ public DrillFileSystem(Configuration fsConf) throws IOException {
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
-    this.underlyingFs = FileSystem.get(fsConf);
+    this(fsConf, URI.create(fsConf.getRaw(FS_DEFAULT_NAME_KEY)), operatorStats);
+  }
+
+  public DrillFileSystem(Configuration fsConf, URI Uri, OperatorStats operatorStats) throws IOException {
```

Comment:
This constructor is added to allow passing a URI. But, the rules for `FileSystem` (see above), is that the URI is stored in the config as a property. So, is this constructor needed? (One argument for using it is that the base class, `FileSystem`, does provide this form.)

Also, this constructor is new. But there is no code in this PR that calls this constructor. So, how does this new (unused) constructor fix the bug? Perhaps a bit of explanation would clear up the mystery...

Reply:
Removed this constructor